core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/HDFSFileDeleteAction.java

```
@@ -0,0 +1,115 @@
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * Action that deletes file(s) within HDFS in the same cluster.
+ * A user must specify file/directory path.
+ * Optionals include fileRegex
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name("HDFSFileDeleteAction")
+@Description("Action to delete files on HDFS")
+public class HDFSFileDeleteAction extends Action {
+  private static final Logger LOG = LoggerFactory.getLogger(HDFSFileDeleteAction.class);
+
+  private HDFSDeleteActionConfig config;
+
+  public HDFSFileDeleteAction(HDFSDeleteActionConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    Path path = new Path(config.path);
+
+    FileSystem fileSystem = path.getFileSystem(new Configuration());
+
+    FileStatus[] listFiles;
+    if (config.fileRegex != null) {
+      PathFilter filter = new PathFilter() {
+        private final Pattern pattern = Pattern.compile(config.fileRegex);
+
+        @Override
+        public boolean accept(Path path) {
+          return pattern.matcher(path.getName()).matches();
+        }
+      };
+      listFiles = fileSystem.listStatus(path, filter);
+    } else {
+      listFiles = fileSystem.listStatus(path);
+    }
+
+    for (FileStatus file: listFiles) {
+      path = file.getPath();
+      if (!fileSystem.delete(path, true)) {
+        if (!config.continueOnError) {
+          throw new IOException(String.format("Removal of %s was unsuccessful.", path.toString()));
+        }
+        LOG.warn("Removal of {} was unsuccessful.", path.toString());
+      }
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+
```

if the fileRegex exists, try compiling it here so that an invalid one fails early at pipeline creation rather than failing at runtime. If we're not doing this in the file move action, should do it there too.